sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/controller/container/ThroughputContainerController.java

```diff
@@ -110,37 +108,47 @@ private ThroughputResolveLevel getThroughputResolveLevel(Set<ThroughputControlGr
     @Override
     @SuppressWarnings("unchecked")
     public <T> Mono<T> init() {
-        return this.resolveDatabaseResourceId()
-            .flatMap(controller -> this.resolveContainerResourceId())
-            .flatMap(controller -> this.resolveContainerMaxThroughput())
+        return this.resolveContainerMaxThroughput()
             .flatMap(controller -> this.createAndInitializeGroupControllers())
             .doOnSuccess(controller -> {
                 Schedulers.parallel().schedule(() -> this.refreshContainerMaxThroughputTask(this.cancellationTokenSource.getToken()).subscribe());
             })
             .thenReturn((T) this);
     }
 
-    private Mono<ThroughputContainerController> resolveDatabaseResourceId() {
+    private Mono<String> resolveDatabaseResourceId() {
         return this.targetContainer.getDatabase().read()
             .flatMap(response -> {
                 this.targetDatabaseRid = response.getProperties().getResourceId();
-                return Mono.just(this);
+                return Mono.just(this.targetDatabaseRid);
             });
     }
 
-    private Mono<ThroughputContainerController> resolveContainerResourceId() {
+    private Mono<String> resolveContainerResourceId() {
         return this.targetContainer.read()
```

Comment:
Are you intentionally not using the CollectionCache from RxCLientImpl?

Reply:
Yea, I was originally thinking using container.read so to get the update to date collection info, in case we have staled info in the cache. hmm, but not sure whether this is a good decision.

Reply:
I think the overhead here isn't worth it. COntainer read goes against master resource budget - very constraint resource. I would use the cache instead like we do for FeedRanges (FeedRangeInternal.getEffectiveXXX etc.)

Reply:
Worst case you will incorrectly calculate the clinet side enforced RU budget after customers delete and recreate a collection - that is a scenario that could happen in unit tests - but not something I would worry about in the real world. So I think it is perfectly safe and a better trade-off to stick to cached results.

Reply:
Make sense, will update to use collection cache in the next iteration.